src/pipeline_edsl/to_workflow.ml

```diff
@@ -529,14 +667,26 @@ module Make (Config : Compiler_configuration)
         ~name:(sprintf "Input-bam: %s" sample_name)
         ~edges:[depends_on f]
     )
+    |> AF.with_provenance "bam" ["file", AF.get_provenance input]
```

Comment:
We have been using `Bfx.file` for a while. 
It is usually hidden by the higher-level functions of the edsl-library:
https://github.com/hammerlab/biokepi/blob/provenance-on-steroids/src/pipeline_edsl/pipeline_library.mli#L91